app/controllers/application_controller.rb

```diff
@@ -1,3 +1,13 @@
 class ApplicationController < ActionController::Base
   protect_from_forgery with: :exception
+
+  def require_user
+    redirect_to sign_in_path unless current_user
+  end
+
+  def current_user
+    User.find(session[:user_id]) if session[:user_id]
```

Comment:
Well this definitely isn't bad, but you can do better than this. Try to use the knowledge from course2 and memoization.

HOMEWORK: watch [this Railscast](http://railscasts.com/episodes/1-caching-with-instance-variables) and try to understand why `current_user` should be memoized. Learn more about memoization.